### PR TITLE
[SCV-22][SCV-26] Validate STAC

### PR DESCRIPTION
**Pull request by quetcodesfire**
These tickets brought together two goals, validate stac items, and return 400 on invalid items. In the past, you could search for invalid collections and providers that didn't exist and get a "bare-bones" stac collection. There was also an issue with features that missed some required data coming in as well.

My solution was to add functions to the validator that would validate stac items. I ultimately ended up splitting the schema validator functions and stac validator functions into two files and exporting them through `index.js` after a suggestion from @DrewE84. I also created a test for the new functions. In addition to this, I responded to invalid stac search items returning `400` responses.

### CODE REVIEW

File: search/lib/validator/stacValidator.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+function validateStac (stacItem) {
+  if (!stacItem) throw new Error('Missing stacItem');
+  try {
+    createStacValidator(stacItem);
+  } catch (e) {
+    return false;
+  }
+
+  return true;
+}
```
Comment (DrewE84) [inline search/lib/validator/stacValidator.js:14]: I feel like these two functions could be combined into a single `validateStac()` function. The current `validateStac()` isn't really doing anything except calling `createStacValidator()`.

File: search/tests/validator/validator.spec.js [hunk 2]
```diff
@@ -4,7 +4,9 @@ const {
   loadOpenApiYaml,
   getSchema,
   getSchemaCollection,
-  validateSchema
+  validateSchema,
+  createStacValidator,
+  validateStac
```
Comment (DrewE84) [inline search/tests/validator/validator.spec.js:9]: This file should probably be split into two test files, one for each file being tested, to fit with the convention in the rest of the application. I.e. a `schemaValidator.spec.js` file and a `stacValidator.spec.js` file.

File: search/lib/api/stac.js [hunk 3]
```diff
@@ -18,14 +21,27 @@ async function getSearch (request, response) {
...
   const params = cmr.convertParams(cmr.STAC_QUERY_PARAMS_CONVERSION_MAP, request.query);
   const result = await search(event, params);
-  response.status(200).json(result);
+  validResult = validateStac(result);
+
+  if (validResult) {
+    response.status(200).json(result);
+  } else {
+    response.status(400).json('Bad Request');
+    console.log('json test', response.status(400).json());
```
Comment (DrewE84) [inline search/lib/api/stac.js:30]: This console.log should be removed before going into production.

**Comment by quetcodesfire** [comment]
Everything looks good to go.

**Pull request merged**